falcon/response.py

```diff
@@ -384,6 +384,20 @@ def set_header(self, name, value):
         # NOTE(kgriffs): normalize name by lowercasing it
         self._headers[name.lower()] = value
 
+    def delete_header(self, name):
+        """Delete a header for this response.
+
+        If the header was not previously set, do nothing.
+
+        Args:
+            name (str): Header name (case-insensitive).  Must be of type
+                ``str`` or ``StringType`` and contain only US-ASCII characters.
+                Under Python 2.x, the ``unicode`` type is also accepted,
+                although such strings are also limited to US-ASCII.
+        """
+        # NOTE(kgriffs): normalize name by lowercasing it
+        self._headers.pop(name.lower(), None)
```

Comment:
Out of curiosity I benchmarked this approach against wrapping a `del` within a `try...except` block and the latter was slightly more performant, but only by a few ns, so it's nothing to worry about.